controllers/clusterenvironment_controller.go

```diff
@@ -278,6 +278,20 @@ func (r *ClusterEnvironmentReconciler) MonitorHealth(ctx context.Context, ns str
 					}
 				}
 			}
+			if cfg == nil {
+				l.Error(errors.New("Empty conig"), "kubeclient config is empty")
+				c := workercluster.ConnectionStatus{
+					State:   primazaiov1alpha1.ClusterEnvironmentStateOffline,
+					Reason:  ErrorDuringHealthCheckReason,
+					Message: "kubeclient config is empty",
+				}
+				r.updateClusterEnvironmentStatus(ctx, ce, c)
+				if err := r.Client.Status().Update(ctx, ce); err != nil {
+					l.Error(err, "error updating cluster environment status", "status", ce.Status)
+				}
+				time.Sleep(time.Duration(hci) * time.Second)
```

Comment:
Is this needed? I'd say we can continue directly with next ClusterEnvironment in the list and sleep at L333

```suggestion
```

Reply:
Done.